GoogleMulticastAppDelegate/Sources/Public/GoogleUtilities/GULMulticastAppDelegate.h

```diff
@@ -0,0 +1,40 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+
+#import <GoogleUtilities/GULApplication.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol GULMulticastAppDelegateProtocol <NSObject>
+
+- (void)addInterceptorWithDelegate:(id<GULApplicationDelegate>)interceptor;
+
+- (void)removeInterceptorWithDelegate:(id<GULApplicationDelegate>)interceptor;
+
+@end
+
+@interface GULMulticastAppDelegate : NSObject <GULApplicationDelegate>
+
+@property(nonatomic, copy) id<GULApplicationDelegate> defaultAppDelegate;
```

Comment:
you mean the GULMulticastAppDelegate and defaultAppDelegate?

it should be weak because it is passed from test app app delegate, so test app should own the app delegate.

Reply:
Yes, I meant that. Alright, thanks, I understand better.

So, should the property be declared like then:
```diff
- @property(nonatomic, copy) id<GULApplicationDelegate> defaultAppDelegate;
+ @property(nonatomic, weak, nullable) id<GULApplicationDelegate> defaultAppDelegate;
```

Reply:
Just realize we need to retain it to keep it in the interceptors list so I think this should be copy.